libs/commands/publish/src/lib/throttle-queue.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * A queue abstraction for tasks that may require a delayed execution
+ */
+export interface Queue {
+  /**
+   * Run a task once the queue is cleared
+   * @param f A function that will be executed once the queue is useable
+   * @return A promise that wraps the value returned by f
+   */
+  queue<T>(f: () => Promise<T>): Promise<T>;
+}
+
+/**
+ * A queue that immediately executes any promise it receives
+ */
+export class ImmediateQueue implements Queue {
```

Comment:
I don't think we need this concept, let's just not leverage the Queue if not throttling

Reply:
The idea is to not have to think about the queue when using it. With this "abstraction", the queue can be used transparently and its instantiation can be delegated to a function no matter what the user-configured value is (it also allows to implement new queue types transparently as this specific implementation seems to work well-enough for my need with NPM but other environments/users might have different needs).

Reply:
For sure, I get it, it's just one more abstraction and one more moving piece to be tested, and in this case it is modifying the existing common case of not needing throttling. I'd like to keep that path unchanged but still support your case